optimize/backend/pom.xml

```diff
@@ -217,11 +248,13 @@
       <artifactId>spring-security-oauth2-resource-server</artifactId>
       <version>${spring.security.version}</version>
     </dependency>
+    <!-- unused/declared
     <dependency>
       <groupId>com.nimbusds</groupId>
       <artifactId>nimbus-jose-jwt</artifactId>
       <version>${nimbus.jose.jwt.version}</version>
     </dependency>
+    -->
```

Comment:
Hmm this might be used for C8SaaS auth. We wouldn't have an automated test for this case, but we should at least test it manually to be sure we aren't breaking SaaS auth